Python/Tests/DebuggerTests/AttachTests.cs

```diff
@@ -453,18 +453,26 @@ import sys
             try {
                 using (var dumpWriter = new MiniDumpWriter(p)) {
                     Thread.Sleep(1500);
+                    Assert.IsFalse(p.HasExited);
 
-                    var attached = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
-                    var bpHit = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+                    var attached = new TaskCompletionSource<bool>();
```

Comment:
We have a thing called `EventTaskSource`. It is in tests now, but can be moved here. Its purpose is to safely await for the event to fire once after subscription with continuation on the thread the event has fired.

Reply:
Looks like it needs to be extended to also allow cancellation by a different event. Can't guarantee I'll get to that extra work anytime soon.

Reply:
Cancellation? Meaning that if one event happens before another test should fail?

Reply:
That's a good point - we can just fail the test on the other event :)

But I tried switching this over to the EventTaskSource (ignoring the other event) and the deadlock came back...